docs/pywbemcli/cmdlineinterface.rst

```diff
@@ -529,3 +581,174 @@ Pywbemcli terminates with one of the following program exit codes:
+
+The following table shows the eval statement for several shells to initialize
+tab completion that would be added to a shell startup file defined in the
+table.
+
+.. table:: Eval statement and proposed startup shell startup file to use for several shells
+   :name: auto-complete-eval-statement
+
+  ======  =======================================  =============================================================
+  Shell   File to insert eval statement            Eval command
+  ======  =======================================  =============================================================
+  bash    ~/.bashrc                                eval "$(_PYWBEMCLI_COMPLETE=bash_source pywbemcli)"
+  zsh     ~/.zshrc                                 eval "$(_PYWBEMCLI_COMPLETE=zsh_source pywbemcli)"
+  fish    ~/.config/fish/completions/foo-bar.fish  eval (env _PYWBEMCLI_COMPLETE=fish_source pywbemcli)
+  ======  =======================================  =============================================================
+
+The above method has some limitations. It may be difficult when virtual
+environments are used and the location of the pywbemcli executable is not known
+until a virtual environment is activated since the eval statement calls
+pywbemcli to get the script itself. It can slow down terminal startup because
+pywbemcli must be called on each terminal startup to get the completion script
+definition
+
+Activation by creating a completion script
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+An alternative is to create a completion script using the same basch statement
+ (ex. ``_PYWBEMCLI_COMPLETE=bash_source pywbemcli``) but
+saving the resulting script in a file.  This that can be executed when
+pywbemcli is to be used and its location is known to create a completion script
+file in a known location for pywbemcli. A file containing shell commands is
+created that defines the completion and completion activation and the tab
+compietion is then activated by executing it (``source ~/.pywbemcli-complete.bash``).
+
+.. table:: Creation of the completion script as the tab completion initialization
+   :name: Shell completion script
+
+  =====  ==========================================================================
+  Shell  Script
+  =====  ==========================================================================
+  bash   _PYWBEMCLI_COMPLETE=bash_source pywbemcli > ~/.pywbemcli-complete.bash
+  zsh    _PYWBEMCLI_COMPLETE=zsh_source pywbemcli > ~/.pywbemcli-complete.zsh
+  fish   _PYWBEMCLI_COMPLETE=fish_source pywbemcli > ~/.config/fish/completions/pywbemcli.fish
+  =====  ==========================================================================
+
+The activation of the completion script can be done either by including the
+activation statement in a terminal startup script (ex. .bashrc) or simply by
+executing the script itself(ex. ``source  ~/.pywbemcli-complete.bash``).
+
+Thus, for bash, pywbemcli can be activated by inserting the following
```

Comment:
This is a repetition of what was stated above.

Reply:
Cleaned up